netlink-sys/src/addr.rs

```diff
@@ -64,7 +64,7 @@ use std::{
 /// The application can also pick the port number by calling Socket::bind with an address with a non-zero port
 /// number. However, it must ensure that this number is unique for each socket created. For instance:
 ///
-/// ```rust
+/// ```rust,should_panic
```

Comment:
mmh this is not supposed to panic, do you mind sharing the error you get?

Reply:
Hm I think this causes an error about "address already in use", doesn't it?

https://github.com/little-dude/netlink/blob/d799194a13c653c440652e1fe44e03a7f72b03bb/netlink-sys/src/addr.rs#L79-L82

Reply:
Yes, the "address already in use" is expected, but that's why we `unwrap_err` actually.

```rust
socket2.bind(&addr2).unwrap_err(); 
```

My only hypothesis is that the first `bind` could trigger this error, if we have another test running in parallel that uses the same address. Could you try with `cargo test -- --test-threads=1`?

Reply:
Oh, I totally missed `unwrap_err`, sorry!
So after some debugging, the `test-threads` flag didn't work well (https://github.com/JohnTitor/netlink/runs/1240889137?check_suite_focus=true), and using socket number "2" worked as expected (https://github.com/JohnTitor/netlink/runs/1240969349?check_suite_focus=true). Uhm, generally, I'd avoid changing docs just only for CI but it's quite minor. If you don't want to change it, I'd add the `--lib` flag to exclude doctests until we find a better way.

Reply:
Thanks for all the debugging. I think changing the docs here is fine. I'm a little worried about not understanding where the failure comes from but I'll do some more debugging later.

Reply:
Okay, removed all the debug commits and this is now ready to merge :)

> I'm a little worried about not understanding where the failure comes from but I'll do some more debugging later. It passes on my local machines (nixos and archlinux) and on Travis though, so maybe Github Actions just do something weird.

The failure comes from the first `bind`'s unwrap so I guess the port number is used on GHA anyways :thinking: